### PR TITLE
WIP Magic Fields

### LINKED ISSUE #234: Magical Field System
Need a way to do fields such that:

* Allow conversions/castings when set
* Allow virtual/generated fields
* Allows values to be set in the class definition of a subclass

### PR DESCRIPTION
**Pull request by astronouth7303**
Implement a field system, to try to remove warts around things like:

```
class MySprite(BaseSprite):
    position = 0, 1
```

```
class MySprite(BaseSprite):
    facing = 1, 1
```

```
class MySprite(BaseSprite):
    left = -4
```

Closes #234

### CODE REVIEW

File: ppb/fields.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    Converts an annotations dict into a fields dict
+    """
+
+
+class FieldMixin:
+    """
+    Mixin that implements all the field magic
+    """
+    def __init_subclass__(cls, **_):
+        # If new fields are defined, make a __fields__
```
Comment (pathunstrom) [inline ppb/fields.py]: Should we worry about the lack of super call?

Comment (astronouth7303) [reply ppb/fields.py]: I probably just forgot about it.

**Comment by astronouth7303** [comment]
I think this is ready for the first round of comments.

In particular, I'm worried test coverage isn't good enough.

**Comment by astronouth7303** [comment]
Oh, look, `ppb_mutant.index` does change things on the class level and expects that to work.

_sigh_ I'll have to implement that. Will probably refactor everything into a metaclass.

**Comment by astronouth7303** [comment]
Since Piper and I both agree this might be a step _too_ magical, I'm going to close this PR.

**Pull request closed without merging**